src/Engine/Manager/VRManager.cpp

```diff
@@ -94,24 +136,45 @@ glm::mat4 VRManager::GetControllerPoseMatrix(int controlID) const {
         if (role == vr::ETrackedControllerRole::TrackedControllerRole_Invalid)
             continue;
         else if (role == vr::ETrackedControllerRole::TrackedControllerRole_LeftHand && controlID == 1) {
-            return glm::inverse(deviceTransforms[untrackedDevice]);
+            return deviceTransforms[untrackedDevice];
         }
         else if (role == vr::ETrackedControllerRole::TrackedControllerRole_RightHand && controlID == 2) {
-            return glm::inverse(deviceTransforms[untrackedDevice]);
+            return deviceTransforms[untrackedDevice];
         }
     }
 
     return glm::mat4();
 }
 
-glm::mat4 VRManager::GetHMDEyeToHeadMatrix(vr::Hmd_Eye eye) const {
+glm::mat4 VRManager::GetHMDHeadToEyeMatrix(vr::Hmd_Eye eye) const {
     if (vrSystem == nullptr) {
         Log() << "No initialized VR device.\n";
         return glm::mat4();
     }
     return glm::inverse(ConvertMatrix(vrSystem->GetEyeToHeadTransform(eye)));
 }
 
+glm::mat4 VRManager::GetHandleTransformation(int controlID, Entity* entity) {
+    glm::mat4 ctrlTransform = GetControllerPoseMatrix(controlID);
+    glm::vec3 ctrlRight = glm::vec3(ctrlTransform[0][0], ctrlTransform[1][0], ctrlTransform[2][0]);
+    glm::vec3 ctrlUp = glm::vec3(ctrlTransform[0][1], ctrlTransform[1][1], ctrlTransform[2][1]);
+    glm::vec3 ctrlForward = glm::vec3(ctrlTransform[0][2], ctrlTransform[1][2], ctrlTransform[2][2]);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'SetScale' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)